app/services/wallets/validate_service.rb

```diff
@@ -43,5 +44,27 @@ def valid_granted_credits_amount?
 
       add_error(field: :granted_credits, error_code: 'invalid_granted_credits')
     end
+
+    def valid_recurring_transaction_rules?
+      if args[:recurring_transaction_rules].count != 1
+        return add_error(field: :recurring_transaction_rules, error_code: 'invalid_number_of_recurring_rules')
+      end
+
+      recurring_rule = args[:recurring_transaction_rules].first
+
+      if recurring_rule[:rule_type]&.to_s == 'interval' &&
+        RecurringTransactionRule.intervals.keys.include?(recurring_rule[:interval])
```

Comment:
Performance/InefficientHashSearch: Use `#key?` instead of `#keys.include?`. (https://github.com/JuanitoFatas/fast-ruby#hashkey-instead-of-hashkeysinclude-code)